### PR TITLE
Add Reverse Bits Algo in Bit-Manipulation

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
Hi, @siriak Check!

File: src/test/java/com/thealgorithms/bitmanipulation/ReverseBitsTest.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class ReverseBitsTest {
+
+    @Test
+    void testReverseBits() {
+        assertEquals(0, ReverseBits.reverseBits(0));
```
Comment (siriak) [inline src/test/java/com/thealgorithms/bitmanipulation/ReverseBitsTest.java:11]: Why should it be 0? All ones is -1 in two's complement

Comment (BamaCharanChhandogi) [reply src/test/java/com/thealgorithms/bitmanipulation/ReverseBitsTest.java:11]: Actually, Negative numbers are stored by taking the bitwise complement of the positive counterpart's binary representation and then adding 1 to the result.
If we want to see a meaningful binary representation of the reversed bits for negative numbers, you could consider using a more complex approach that handles negative numbers differently.
 
should I remove negative Test case?

Comment (siriak) [reply src/test/java/com/thealgorithms/bitmanipulation/ReverseBitsTest.java:11]: My bad, for whatever reason I thought that your function would invert bits, not reverse their order.

**Review by siriak — APPROVED** [review]

**Pull request merged**